src/dxtbx/nexus/nxmx.py

```diff
@@ -113,7 +113,8 @@ def __iter__(self):
         return iter(self._handle)
 
     def __len__(self):
-        return len(self._handle)
+        shape = self._handle.shape
+        return shape[0] if shape else self._handle.size
```

Comment:
@rjgildea, I would particularly welcome your eyes on this, in case I'm not holding `self._handle` correctly.

Reply:
I think the only time it's not a `h5py.Group` or `h5py.File` is when `type(self)` is `dxtbx.nexus.nxmx.NXtransformationsAxis`. What is this use case is this change supposed to be handling?

Reply:
Looks like this is resolved by https://github.com/cctbx/dxtbx/pull/428/commits/a75c97a5177799af5574fe7b2b4081f5d6d6c25c.